### ISSUE TITLE
Missing ffmpeg filters (select)

### ISSUE THREAD

**Issue by arrietaeguren** [wang-bin/QtAV#714]
Heya, first of all, many many thanks for developing this amazing project.

the [select filter](http://www.ffmpeg.org/ffmpeg-filters.html#select_002c-aselect) available in ffmpeg is not supported in QtAV :( I really need that filter in the project I'm building. Is there any way to enable it? Are you planning to support it? When? Is there any way I can help in the process? Any tips on where to dig in the code to enable it?

Thanks

**Comment by wang-bin** [comment]
What's your filter code? QtAV's Audio/VideoFrame misses many attributes when converting from ffmpeg AVFrame, so not all filters are supported

**Comment by arrietaeguren** [comment]
Thanks for your reply. The ffplay equivalent of what I want to do would be:

```
ffplay
-vf "select='lte(t\,4)+gte(t\,16)',setpts=N/FRAME_RATE/TB"
-af "aselect='lte(t\,4)+gte(t\,16)',asetpts=N/SR/TB"
-i INPUT
```

which is basically skipping the selected clip. On qml I've tried the example, replacing the filter
```
import QtAV 1.7
...
VideoFilter {
    id: vf
    avfilter: "select='lte(t\,4)+gte(t\,16)',setpts=N/FRAME_RATE/TB"
    ...
}
...
MediaPlayer {
    ...
    videoFilters: [vf]
}
```
Is there any possible work around to apply this filter? I like your project and I don't want to drop it

**Comment by wang-bin** [comment]
I have to change some code. BTW, what does `N/FRAME_RATE/TB` mean?

**Comment by arrietaeguren** [comment]
Could you do it? It would be awesome and I would love you forever!!

The `select` filter removes the selected frames. The other frames would still be presented when marked by their corresponding presentation time stamp (PTS), leaving a horrible black gap during playback. `setpts` solved this by changing the PTS of the output frames:

`N`: stands for the number of consumed samples. This count ignores the frames skipped by `select`. It's the frame number on the output video
`FRAME_RATE`: the number of frames per second. (`SR` is the same but for the audio)
`TB`: is the timebase of the input time stamps. The pts time is not always measured in seconds, it could be measured in milliseconds, 1/4000 of a second or whatever. This is just a scale factor to change the time units.

So basically what I'm doing is modifying the PTS of the output video, so that frames are presented continuously on the output. The new PTS in seconds is just `"FRAME_NUMBER"/FRAME_RATE`

The [ffmpeg docs](http://www.ffmpeg.org/ffmpeg-filters.html#setpts_002c-asetpts) on `setpts` might also help you
